src/_pytest/assertion/util.py

```diff
@@ -436,17 +451,25 @@ def _compare_eq_cls(
     if same and verbose < 2:
         explanation.append("Omitting %s identical items, use -vv to show" % len(same))
     elif same:
-        explanation += ["Matching attributes:"]
-        explanation += pprint.pformat(same).splitlines()
+        explanation += ["{}Matching attributes:".format(indent)]
+        lines = pprint.pformat(same).splitlines()
+        for line in lines:
+            explanation += [indent + str(line)]
     if diff:
-        explanation += ["Differing attributes:"]
+        explanation += ["{}Differing attributes:".format(indent)]
         for field in diff:
+            indent_level += 1
```

Comment:
This one is not exactly right; it causes differing attributes at the same tree level to be indented differently.

Generally since the indentation is only done one level now, I think we can do away with the `indent_level` variable and just add the indentation directly.

Reply:
I got your point - sorry for not doing enough tests, I was sure, that it is necessary to increase `indent_level` by one